packages/nextjs/app/builders/page.tsx

```diff
@@ -0,0 +1,134 @@
+  args?: {
+    builder?: string;
+    checkInContract?: string;
+    first?: boolean;
+  };
+}
+
+function Members() {
+  const {
+    data: events,
+    isLoading,
+    error,
+  } = useScaffoldEventHistory({
+    contractName: "BatchRegistry",
+    eventName: "CheckedIn",
+    fromBlock: 131801148n,
+  });
+
+  const builders = useMemo(() => {
+    if (!events || !Array.isArray(events)) {
+      console.error("Expected events to be an array, but got:", typeof events, events);
+      return [];
+    }
+
+    return events
+      .filter((event: Event) => event.args?.first === true)
+      .map((event: Event) => ({
+        builder: event.args!.builder as string,
+        checkInContract: event.args!.checkInContract as string,
+      }));
+  }, [events]);
+
+  return (
+    <div className="max-w-6xl mx-auto">
+      <div className="bg-white/10 backdrop-blur-lg border border-white/20 rounded-3xl p-6 shadow-lg">
+        <h2 className="text-3xl font-bold text-center mb-4 text-blue-300">Check-Ins</h2>
+
+        {isLoading ? (
+          <p className="text-center text-gray-400">Loading events...</p>
+        ) : error ? (
+          <p className="text-center text-red-500">Error: {error.message}</p>
+        ) : builders.length > 0 ? (
+          <div className="overflow-x-auto">
+            <table className="w-full border border-gray-700 rounded-lg table-auto">
+              <thead className="bg-gray-700 text-gray-300">
+                <tr>
+                  <th className="px-4 py-3 text-left">Builder Address</th>
+                  <th className="px-4 py-3 text-left">Check-In Contract</th>
+                  <th className="px-4 py-3 text-center">Profile</th>
+                </tr>
+              </thead>
+              <tbody>
+                {builders.map((builderData, index) => (
+                  <BuilderRow key={index} builderData={builderData} />
+                ))}
+              </tbody>
+            </table>
+          </div>
+        ) : (
+          <p className="text-center text-gray-400">No first-time builders found.</p>
+        )}
+      </div>
+    </div>
+  );
+}
+
+function BuilderRow({ builderData }: { builderData: { builder: string; checkInContract: string } }) {
+  const [pageExists, setPageExists] = useState<boolean | null>(null);
+
+  useEffect(() => {
+    const checkPageExists = async () => {
+      try {
+        const response = await fetch(`/${builderData.builder}`);
+        setPageExists(response.ok);
+      } catch (error) {
+        console.error("Error checking page existence:", error);
+        setPageExists(false);
+      }
+    };
+    checkPageExists();
+  }, [builderData.builder]);
+
+  return (
+    <tr className="border-b border-gray-600 hover:bg-gray-700 transition-colors duration-200">
+      <td className="px-4 py-3 font-mono break-all">
+        <Address address={builderData.builder} />
+      </td>
+      <td className="px-4 py-3 font-mono break-all">{builderData.checkInContract}</td>
```

Comment:
As I mentioned in a previous review, this column is too squeezed on mobile devices.  
Maybe we don't need to show the "Check-In Contract."